File: helloworld_web_war/src/main/java/org/example/demo/dso/service/impl/UserServiceImpl.java
package org.example.demo.dso.service.impl;

import org.noear.solon.annotation.Component;

import org.example.demo.dso.service.UserService;
import org.example.demo.model.UserModel;

@Component
public class UserServiceImpl implements UserService {
    @Override
    public UserModel getUser(int userId) {
        UserModel user = new UserModel();
        user.setUserId(userId);
        user.setUserName("user-" + user);

        return user;
    }

    @Override
    public void addUser(UserModel user) {

    }
}

File: helloworld_web_jar_kt/src/main/kotlin/com/example/demo/dso/service/impl/UserServiceImpl.kt
package com.example.demo.dso.service.impl

import com.example.demo.dso.service.UserService
import com.example.demo.model.UserModel
import org.noear.solon.annotation.Component
import org.noear.solon.data.annotation.Tran


@Component
open class UserServiceImpl : UserService {
    override fun getUser(userId: Int): UserModel? {
        val user = UserModel()
        user.userId = userId;
        user.userName = "user-$userId";
        return user
    }

    @Tran
    override fun addUser(user: UserModel?) {}
}

